conformance/tests/httproute-redirect-port.go

```diff
@@ -43,57 +43,97 @@ var HTTPRouteRedirectPort = suite.ConformanceTest{
 		gwAddr := kubernetes.GatewayAndHTTPRoutesMustBeAccepted(t, suite.Client, suite.TimeoutConfig, suite.ControllerName, kubernetes.NewGatewayRef(gwNN), routeNN)
 		kubernetes.HTTPRouteMustHaveResolvedRefsConditionsTrue(t, suite.Client, suite.TimeoutConfig, routeNN, gwNN)
 
-		testCases := []http.ExpectedResponse{{
-			Request: http.Request{
-				Path:             "/port",
-				UnfollowRedirect: true,
+		testCases := []http.ExpectedResponse{
+			{
+				Request: http.Request{
+					Path:             "/port",
+					UnfollowRedirect: true,
+				},
+				Response: http.Response{
+					StatusCode: 302,
+				},
+				RedirectRequest: &roundtripper.RedirectRequest{
+					Port: "8083",
+				},
+				Namespace: ns,
 			},
-			Response: http.Response{
-				StatusCode: 302,
+			{
+				Request: http.Request{
+					Path:             "/port-and-host",
+					UnfollowRedirect: true,
+				},
+				Response: http.Response{
+					StatusCode: 302,
+				},
+				RedirectRequest: &roundtripper.RedirectRequest{
+					Host: "example.org",
+					Port: "8083",
+				},
+				Namespace: ns,
 			},
-			RedirectRequest: &roundtripper.RedirectRequest{
-				Port: "8083",
+			{
+				Request: http.Request{
+					Path:             "/port-and-status",
+					UnfollowRedirect: true,
+				},
+				Response: http.Response{
+					StatusCode: 301,
+				},
+				RedirectRequest: &roundtripper.RedirectRequest{
+					Port: "8083",
+				},
+				Namespace: ns,
 			},
-			Namespace: ns,
-		}, {
-			Request: http.Request{
-				Path:             "/port-and-host",
-				UnfollowRedirect: true,
+			{
+				Request: http.Request{
+					Path:             "/port-and-host-and-status",
+					UnfollowRedirect: true,
+				},
+				Response: http.Response{
+					StatusCode: 302,
+				},
+				RedirectRequest: &roundtripper.RedirectRequest{
+					Port: "8083",
+					Host: "example.org",
+				},
+				Namespace: ns,
 			},
-			Response: http.Response{
-				StatusCode: 302,
+			{
+				// Port should not be expected in the Location header of redirect
+				// response since:
+				// 1. Scheme is defaulted to 'http', and
+				// 2. Redirect port is specified as 80, which is redundant when paired
+				// 		with Scheme.
+				Request: http.Request{
+					Path:             "/port-80",
```

Comment:
It would be ideal to have all 4 combos of {http,https}x{80,443}

Reply:
Definitely. Changed.

Reply:
Also if we can add with `Host: IP` and `Host: IP:port` could be helpful (saying this as an implementation that is broken for one but not the other)